docs_src/security/Ch-Configuring-Add-On-Services.md

```diff
@@ -1 +1,189 @@
+    command: "/app-service-configurable ${DEFAULT_EDGEX_RUN_CMD_PARMS}"
+    volumes:
+      - edgex-init:/edgex-init:ro,z
+      - /tmp/edgex/secrets/app-rules-engine:/tmp/edgex/secrets/app-rules-engine:ro,z
+    depends_on:
+      - security-bootstrapper
+...
+
+```
+
+## Configure the secret store to use
+
+Make sure the TOML configuration file of add-on service like `device-camera` contains
+the proper `[SecretStore]` section.
+
+Example:
+
+```toml
+[SecretStore]
+Type = 'vault'
+Host = 'localhost'
+Port = 8200
+Path = 'my-service/'
+Protocol = 'http'
+RootCaCertPath = ''
+ServerName = ''
+TokenFile = '/tmp/edgex/secrets/my-service/secrets-token.json'
+  [SecretStore.Authentication]
+  AuthType = 'X-Vault-Token'
+```
+
+and then in the EdgeX service `secretstore-setup` environment section of `Docker-compose` file to add
+the service key or hostname to the environment variable `ADD_SECRETSTORE_TOKENS`:
+
+```yaml
+...
+  secretstore-setup:
+    container_name: edgex-secretstore-setup
+    depends_on:
+    - security-bootstrapper
+    - vault
+    environment:
+      ADD_SECRETSTORE_TOKENS: 'device-camera'
+...
+
+```
+
+With that, `secretstore-setup` then can generate secretstore token from `Vault` and store it in
+the `TokenFile` path specified in the TOML configuration file like the above example.
+
+## Configure the secure Redis message bus to use
+
+This can be done in the EdgeX service `secretstore-setup` environment section of
+`Docker-compose` file to add the `redisdb` and the service key using it.
+
+So an example will be like `redisdb[device-virtual]` in which `redisdb` is the database resource name
+and the `device-virtual` is the service key or hostname and add that value into the environment variable
+`ADD_KNOWN_SECRETS`:
+
+```yaml
+...
+  secretstore-setup:
+    container_name: edgex-secretstore-setup
+    depends_on:
+    - security-bootstrapper
+    - vault
+    environment:
+      ADD_SECRETSTORE_TOKENS: 'device-camera, my-service'
+      ADD_KNOWN_SECRETS: redisdb[app-rules-engine],redisdb[device-rest],redisdb[device-virtual]
+...
+
+```
+
+
+## (Optional) Configure the ACL role of configuration/registry to use if the service depends on it
+
+This is a new step coming from `securing Consul` security features as part of EdgeX Ireland release.
+
+If the add-on service uses `Consul` as the configuration and/or registry service, then we also need to
+configure the environment variable `ADD_REGISTRY_ACL_ROLES` to tell `security-bootstrapper` to generate
+an ACL role for `Consul` to associate with its token.
+
+An example of configuring acl roles of the registry `Consul` for the add-on services
+`device-modbus` and `app-http-export` is like the following:
+
+```yaml
+...
+  consul:
+    container_name: edgex-core-consul
+    depends_on:
+    - security-bootstrapper
+    - vault
+    entrypoint:
+    - /edgex-init/consul_wait_install.sh
+    environment:
+      ADD_REGISTRY_ACL_ROLES: app-http-export,device-modbus
+...
+
+```
+
```

Comment:
It might be useful to describe what the user might observe if they were supposed to do this but didn't.

Reply:
good point, i'll add that

Reply:
added